src/main/java/org/takes/facets/auth/social/PsTwitter.java

```diff
@@ -64,57 +71,112 @@
+        this.app = tapp;
+        this.key = tkey;
     }
 
     @Override
     public Opt<Identity> enter(final Request request)
         throws IOException {
-        return new Opt.Single<Identity>(PsTwitter.fetch(this.token()));
+        return new Opt.Single<Identity>(this.fetch(this.token()));
     }
 
     @Override
-    public Response exit(final Response response,
-        final Identity identity) {
+    public Response exit(final Response response, final Identity identity) {
         return response;
     }
 
     /**
-     * Get user name from Twitter, with the token provided.
-     * @param token Twitter access token
+     * Get user name from Twitter, with the trequest provided.
+     * @param token Twitter access trequest
      * @return The user found in Twitter
      * @throws IOException If fails
      */
-    private static Identity fetch(final String token) throws IOException {
-        final String uri = new Href(
-            "https://api.twitter.com/1.1/account/verify_credentials.json"
-        ).with("access_token", token).toString();
-        return PsTwitter.parse(
-            new JdkRequest(uri)
+    private Identity fetch(final String token) throws IOException {
+        return parse(
+            this.user
+                .uri()
+                .set(
+                    URI.create(
+                        new Href(PsTwitter.VERIFY_URL)
+                            .with("access_token", token)
+                            .toString()
+                    )
+                )
+                .back()
                 .header("accept", "application/json")
                 .fetch().as(RestResponse.class)
                 .assertStatus(HttpURLConnection.HTTP_OK)
-                .as(JsonResponse.class).json().readObject()
+                .as(JsonResponse.class)
+                .json()
+                .readObject()
         );
     }
 
     /**
-     * Retrieve Twitter access token.
-     * @return The token
+     * Make identity from JSON object.
+     * @param json JSON received from Twitter
+     * @return Identity found
+    */
+    private static Identity parse(final JsonObject json) {
+        final ConcurrentMap<String, String> props =
+                new ConcurrentHashMap<String, String>(json.size());
+        props.put("name", json.getString("name"));
+        props.put("picture", json.getString("profile_image_url"));
+        return new Identity.Simple(
+            String.format("urn:twitter:%d", json.getInt("id")),
+            props
+        );
+    }
+
+    /**
+     * Retrieve Twitter access trequest.
+     * @return The trequest
      * @throws IOException If failed
      */
-    private String token()
-        throws IOException {
-        final String uri = new Href("https://api.twitter.com/oauth2/token")
-            .with("grant_type", "client_credentials")
-            .toString();
-        return new JdkRequest(uri)
+    private String token() throws IOException {
```

Comment:
@adko-pl let's rename this to `fetchToken`. that way we free up the name `token` for our variable above.